bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -68,9 +69,16 @@
     long lastMinorCompactionTime;
 
     boolean enableMajorCompaction = false;
-    final double majorCompactionThreshold;
+    final double originalMajorCompactionThreshold;
+    double majorCompactionThreshold;
```

Comment:
I don't think you want to maintain both majorCompactionThreshold and originalMajorCompactionThreshold.  majorCompactionThreshold is mainly used in two places:
1. In the constructor, you do some validation of the config, but you don't need to change it to do that.
2. In runWithFlags, you set it based on the current time, pass it into doCompactEntryLogs, and restore it back to original before returning.

Instead, rename originalMajorCompactionThreshold to majorCompactionThreshold (and leave it final) and get rid of the mutable majorCompactionThreshold.  In runWithFlags, modify changeMajorCompactionThreshold to getCurrentMajorCompactionThreshold and let it return a value based on the current time which you can keep in a local in runWithFlags.  You don't need restoreThreshold at all.

Reply:
good idea, thanks.